public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -424,15 +427,25 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                 if(!conceptSynonymNames.contains(conceptSynonym.getConceptSynonymName()) && !con.getConceptName().equals(conceptSynonym.getConceptSynonymName())){
                     conceptSynonymNames.add(conceptSynonym.getConceptSynonymName());
                     }
-            }
 
+            }
 
             this.conceptSynonymNames.put(con.getConceptId(),conceptSynonymNames);
 
             if((con.getStandardConcept() == null || !con.getStandardConcept().equals("S") ) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))){
                 response.setMatchType(conceptCode.equals(searchConceptsRequest.getQuery()) ? MatchType.CODE : MatchType.ID );
 
                 List<Concept> std_concepts = conceptDao.findStandardConcepts(con.getConceptId());
+                for(Concept concept: std_concepts){
```

Comment:
I think you should update the conceptDao.findStandardConcepts to fetch the synonyms since we have all the relationship setup and everything and it's not a lot of rows. Then you don't need this code.

Reply:
It was not possible to join fetch with concept relationship and concept synonym at the same time and the concept synonyms were not fetched by default as we specified it to be lazy. So added a query in dao to join fetch concept synonyms of only standard concept by id.

Reply:
Hmm.. Ok. In that case i would use this is ConceptService ...
https://github.com/all-of-us/workbench/blob/master/common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java#L230